DEADMAN_155/Deadman_Check.py

```diff
@@ -0,0 +1,370 @@
+        eqf013m_acq = False
+        eef1000_acq = False
+
+    # If SCS-155 is enabled and activated, and if you have acquired the OORMPDS time
+    # after the 155 enable and activate, then you can check the time interval between
+    # SCS-155 enable and OORMPDS.
+    if scs_155_en and scs_155_act and oormpds_acq:
+        # Set the required secondss between SCS-155 enable and OORMPDS
+        required_minutes = 2880.0
+        required_seconds = (48.0 * 3600.0)
+
+        delta_t_seconds, delta_t_minutes, delta_t_hours = cd.Calc_Delta_Time(COENAS1_date, OORMPDS_date)
+        # Check the value returned - it should be 48 hours
+        required_v_actual = required_seconds - delta_t_seconds
+
+        # If the time delta between SCS-155 enable and OORMPDS differs by less
+        # than 3 minutes, all is good. Otherwise throw an error
+        if abs(required_seconds - delta_t_seconds) <= 180.0:
+            print('     '+OORMPDS_date+'      RADMON DISABLE  ==> Time between 155 ENABLE and OORMPDS:  %.4f hours off by %.4f seconds.  Ok.' % (delta_t_hours, required_v_actual) )
+        else:
+            print('     '+OORMPDS_date+'      RADMON DISABLE ==> Time between 155 ENABLE and OORMPDS: %.4f hours off by %.4f seconds. ERROR.' % (delta_t_hours, required_v_actual) )
+
+        # Shut off the booleans for enable and activate to avoid recalculation
+        scs_155_en=False
+        scs_155_act=False
+
+    # If you have acquired the OORMPDS time and the SCS-155 DISABLE time
+    # then you can calculate the delta shutoff time between RADMON DIS and 
+    # SCS-155 DISA
+    if oormpds_acq and scs_155_disa:
+        delta_t_sec, delta_t_minutes, delta_t_hours = cd.Calc_Delta_Time(OORMPDS_date, CODISAS1_date)
+        # Check the calculated time delta - it should be 5 minutes (rounded)
+        delta_t_minutes = round(delta_t_minutes, 0)
+        if delta_t_minutes == 5.0:
+            print('     '+CODISAS1_date+'      SCS-155 Disable ==> Time between OORMPDS and SCS-155 DISA', delta_t_minutes, 'minutes. Ok.')
+        else:
+            print('     '+CODISAS1_date+'      SCS-155 Disable >>> ERROR: Time Between OORMPDS and DEACTIVATION is NOT 5 minutes: ', delta_t_minutes, 'minutes' )
+
+        # Calculate the prospective deadman timeout time.
+        # There is a special case where the Continuity load begins with
+        #   EQF013M, CODISAS1=155, and EEF1000
+        # The enable and activate commands appeared in the load before the continuity load.
+        # When this happens, you can do the calculation above which compares only the
+        # RADMON DIS time with the SCS-155 disable time.  But you cannot proceed to
+        # this next section which calculates the time delta between SCS-155 activation
+        # and RADMON DIS (because you don't have the activation time). This is only
+        # a problem at the start analysis of the Continuity load.
+        if COACTS1_time != None:
```

Comment:
This should be `if COACTS1_time is not None:`

Reply:
Changed